File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand_1, strand_2)
    raise ArgumentError if strand_1.length != strand_2.length
    (strand_1.chars).zip(strand_2.chars).count { |pair_of_letters| pair_of_letters[0] != pair_of_letters[1] }
  end

  VERSION = 1
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(strand1, strand2) {
  this.validateLength(strand1.length, strand2.length)

  var counter = 0;
  for (var idx = 0; idx < strand1.length; idx++) {
    if (strand1.charAt(idx) !== strand2.charAt(idx)) {
      counter++;
    };
  };
  return counter;
};

Hamming.prototype.validateLength = function(length1, length2) {
  if (length1 !== length2) {
    throw 'DNA strands must be of equal length.'
  };
}

module.exports = Hamming;